SimulationAgent/SimulationManager.cs

```diff
@@ -228,6 +234,26 @@ public void PrintStats()
                 });
         }
 
+        public async Task SaveStatisticsAsync()
+        {
+            try
+            {
+                var simulationModel = new SimulationStatisticsModel
+                {
+                    TotalMessagesSent = this.deviceTelemetryActors.Where(a => a.Key.StartsWith(simulation.Id)).Sum(a => a.Value.TotalMessagesCount),
+                    FailedMessagesCount = this.deviceTelemetryActors.Where(a => a.Key.StartsWith(simulation.Id)).Sum(a => a.Value.FailedMessagesCount),
```

Comment:
these 2 lines are looping and filtering the same list twice, please merge into one loop